### PR TITLE
Change ForceSSLHandler#secure? to use == instead of regex

### PR DESCRIPTION
**Pull request by grepsedawk**
It's more clear and way faster

Latest benchmark in commit message

### CODE REVIEW

**Comment by grepsedawk** [comment]
Putting this into draft because, while the tests pass, the private method is not exactly the same.
The private method previous was a "#includes?" via regex not `==`.

I'm going to run benchmarks and base my next approach based on those benchmarks.

**Review by matthewmcgarvey — COMMENTED** [review]
I _think_ it will be fine to do `==`. Looking at rails, they delegate to `Rack::Request#ssl?` which if you look through the code branches does the same thing https://github.com/rack/rack/blob/a61c0b4fe7162df76cee617922c332b9b138d425/lib/rack/request.rb#L366-L368

**Comment by grepsedawk** [comment]
Cool, asserting equality is faster and easier anyway
Rack also doesn't _seem_ to downcase the header ever, it boils down to calling: https://github.com/rack/rack/blob/a61c0b4fe7162df76cee617922c332b9b138d425/lib/rack/request.rb#L79 via https://github.com/rack/rack/blob/a61c0b4fe7162df76cee617922c332b9b138d425/lib/rack/request.rb#L643

Unless @env is auto-downcased.

File: src/lucky/force_ssl_handler.cr [hunk 1]
```diff
@@ -48,7 +48,7 @@ class Lucky::ForceSSLHandler
   end
 
   private def secure?(context) : Bool
-    !!(context.request.headers["X-Forwarded-Proto"]? =~ /https/i)
+    context.request.headers["X-Forwarded-Proto"]?.try &.downcase == "https"
```
[the remark this answers is not in the document]
Comment (jwoertink) [reply src/lucky/force_ssl_handler.cr:51]: https://github.com/crystal-lang/crystal/blob/2b1cbee25ff50c0e7b8ef021921e1f295fdbe8d3/src/http/headers.cr#L3

Looks like HTTP::Headers already uses `==`

[the remark this answers is not in the document]
Comment (jwoertink) [reply src/lucky/force_ssl_handler.cr:51]: oh wait.. that's just comparing two headers, and not their value... Ignore me, I'm still waking up lol

**Comment by robacarp** [comment]
The benchmarks are hard to read because you have both the match and mismatch tests in the same group. Can you separate out the two match runs from the mismatch runs so that the `fastest` and `nn x slower` stuff is meaningful?

**Comment by robacarp** [comment]
Re downcasing, I know http header names are supposed to be case insensitive but I don't think I know that about the values. 

The [Rack tests](https://github.com/rack/rack/blob/a61c0b4fe7162df76cee617922c332b9b138d425/test/spec_request.rb#L670-L688) only seem to test that x-forwarded-proto works correctly when it's all lowercase. The [internet wisdom](https://stackoverflow.com/questions/51928504/x-forwarded-proto-https-in-frontend-or-backend-haproxy) about configuring haproxy seems to have decided on using a lowercase `https` but there's no reason for that to be necessarily true.

I think it's fine to remove the downcase.

**Review by matthewmcgarvey — APPROVED** [review]

**Review by robacarp — APPROVED** [review]
awesome 🎉

**Pull request merged**